File: app/src/main/java/demo/victormunoz/githubusers/features/allusers/AllUsersContract.java
package demo.victormunoz.githubusers.features.allusers;

import android.view.View;
import com.trello.rxlifecycle2.android.ActivityEvent;
import java.util.List;
import demo.victormunoz.githubusers.model.entity.User;
import io.reactivex.Observable;

public interface AllUsersContract {

    interface ViewListener {
        void onEndOfTheList();

        void onRetry();

        void onItemClick(View view, User user);

    }

    interface PresenterListener {
        void showError();

        void addUsers(List<User> users);

        void goToUserDetails(View view, String loginName, String avatarURL);

    }

    interface AdapterListener {
        void onEndOfTheList();

        void onItemClick(View view, int position);

        Observable<ActivityEvent> getLifeCycle();
    }

}

File: app/src/main/java/demo/victormunoz/githubusers/features/allusers/AllUsersContract.kt
package demo.victormunoz.githubusers.features.allusers

import android.graphics.Bitmap
import android.view.View
import demo.victormunoz.githubusers.model.User

interface AllUsersContract {

    interface ActivityPresenterListener {

        fun onEndOfTheList()

        fun onRetry()

    }

    interface ViewHolderPresenterListener {

        fun onItemClick(view: View, user: User)

        fun onImageRequest(url:String, size: Int)

    }

    interface ActivityListener {

        fun showError()

        fun showUsers(users: List<User>)

    }
    interface ActivityNavigationListener {

        fun goToUserDetails(view: View, user: User)

    }

    interface ViewHolderListener {

        fun showImage(bitmap:Bitmap)

        fun showError()

    }

    interface AdapterListener {

        fun onEndOfTheList()

    }

}
